### PR TITLE
Cae 119 integration test

### PR DESCRIPTION
**Pull request by rishipisipati**
#### Required
- [x] I ran the CI build script to ensure that the code passes the following
    - Unit tests
    - Integration tests
    - Linting rules
- [x] I ran the E2E test suite locally to ensure that no regressions were introduced

#### Optional
- [ ] My PR contains a DB schema change
    - Reviewers pay special attention to queries/indices when this box is checked
- [ ] My PR is dependent on a change to our infrastructure
    - New SSM keys
    - IAM role updates
    - ETC
- [ ] My pull request is fewer than 400 lines of code

## Purpose

* **What is the context of this pull request? Why is it being done?** **required*
PR for integration test updates on the learning walk workflow.

* **Are there any related PRs? Link to them here.**

* **Link to any additional documentation (Jira, Google Docs, wireframes, etc.)** **required when available*
https://betterlesson.atlassian.net/browse/CAE-119
## Changes

* **How are different categories of users affected by these changes?**

* **Are there any changes that need to be called out as significant or particularly tricky?**

* **What is the recommended reading order for these changes? Where should the reviewer start?**

* **Add screenshots here for any UI changes (try to limit to the most important changes to avoid extending the page too much).**

## Release

* **Are there any special considerations for deployment?** **required*

### CODE REVIEW

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java [hunk 1]
```diff
@@ -248,18 +247,18 @@ private Map<Long, IndicatorResponses> addOrUpdateLearningWalkClassroomSummaryWit
...
                 Collectors.toMap(
                     participantIndicatorResponse -> participantIndicatorResponse.getIndicatorId(),
                     participantIndicatorResponse ->
-                        /* There are two cases where indicators get created instead of being updated
-                        1. The very first response that comes in for a summary and
-                        2. In a scenario where one or more indicatorResponse in an observation
-                           is left unfinished.
-                        We want to add the first response in those cases */
-
                         (!existingResponses.isEmpty()
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java:262]: Extrapolating it into a lambda expression is leading to a Nullaway Exception, despite the fact that we're checking for the key in the if condition on existing responses. Reverted the code back to it's original state. 

See this commit for reference:
https://github.com/BetterLesson/Betterlesson-LearningWalk-API/pull/19/commits/aa89664b507a93f19b4b57688a6c18fecb199e58
(Line 257 in LearningWalkReportingService)

CircleCI: 
https://app.circleci.com/pipelines/github/BetterLesson/Betterlesson-LearningWalk-API/100/workflows/6da5c557-05b3-4612-ac52-4c4fcdd4cae4/jobs/151

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java [hunk 2]
```diff
@@ -248,18 +247,18 @@ private Map<Long, IndicatorResponses> addOrUpdateLearningWalkClassroomSummaryWit
...
     Map<Long, IndicatorResponses> existingResponses = learningWalkClassroomSummary.getResponses();
     List<ParticipantIndicatorResponse> indicatorResponses = learningWalkObservation.getResponses();
+    /* There are two cases where indicators get created instead of being updated
+    1. The very first response that comes in for a summary and
+    2. In a scenario where one or more indicatorResponse in an observation
+       is left unfinished.
+    We want to add the first response in those cases */
     Map<Long, IndicatorResponses> updatedResponses =
         indicatorResponses.stream()
+            .filter(indicatorResponse -> !isIndicatorEmpty(indicatorResponse))
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java:257]: Remove entries with no radio or written responses to be added to the summary.

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java [hunk 3]
```diff
@@ -339,12 +338,16 @@ private void mutateExistingIndicatorSummaryFromParticipantIndicatorResponse(
     String writtenResponse = response.getWrittenResponse();
     Map<String, Long> radioOptionResponses = existing.getRadioOptionResponses();
     List<String> writtenResponses = existing.getWrittenResponses();
-    writtenResponses.add(writtenResponse);
+    if (!writtenResponse.equals("")) {
+      writtenResponses.add(writtenResponse);
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java:342]: Skip adding a written response if nothing has been added yet.

File: src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java [hunk 4]
```diff
@@ -339,12 +338,16 @@ private void mutateExistingIndicatorSummaryFromParticipantIndicatorResponse(
...
+    }
     if (radioOptionResponses.containsKey(radioOptionResponse)) {
       radioOptionResponses.put(
           radioOptionResponse, radioOptionResponses.get(radioOptionResponse) + 1L);
     } else {
-      radioOptionResponses.put(radioOptionResponse, 1L);
+      if (!radioOptionResponse.equals("")) {
+        radioOptionResponses.put(radioOptionResponse, 1L);
```
Comment (rishipisipati) [inline src/main/java/com/betterlesson/learningwalk/learningwalk/services/LearningWalkReportingService.java:349]: Add a radio response only if one was submitted.

File: src/test/java/com/betterlesson/learningwalk/integration/LearningWalkIntegrationTests.java [hunk 5]
```diff
@@ -90,95 +131,195 @@ public void happy_path() {
...
+      when(userService.getUserServiceInfoForParticipant(
+              eq(participants.get(participantNumber)), any()))
+          .thenReturn(betterlessonUserInfo);
+      // Mark participant responses as submitted
+      learningWalkReportingService.submitClassroomObservation(
+          learningWalkResponse.getUuid(), status);
+    }
+
+    // Get classroom summary and assert
+    classroomIds = new ArrayList<>(new LinkedHashSet<>(classroomIds));
```
Comment (nicklundin-bl) [inline src/test/java/com/betterlesson/learningwalk/integration/LearningWalkIntegrationTests.java:261]: Why do we have 2 constructors here?

Comment (rishipisipati) [reply src/test/java/com/betterlesson/learningwalk/integration/LearningWalkIntegrationTests.java:261]: The loop above was adding duplicate values for classroomIds. I suppose I can just use a linkedhashset from the start.

Comment (nicklundin-bl) [reply src/test/java/com/betterlesson/learningwalk/integration/LearningWalkIntegrationTests.java:261]: Yeah do that. Approved

**Pull request merged**